elements/lisk-chain/src/slots.ts

```diff
@@ -13,45 +13,47 @@
  */
 
 interface SlotsInput {
-	readonly epochTime: string;
+	readonly genesisBlockTimestamp: number;
 	readonly interval: number;
 }
 
 const SEC_IN_MS = 1000;
 
 export class Slots {
-	private readonly _epochTime: Date;
+	private readonly _genesisTime: Date;
 	private readonly _interval: number;
 
-	public constructor({ epochTime, interval }: SlotsInput) {
-		this._epochTime = new Date(epochTime);
+	public constructor({ genesisBlockTimestamp, interval }: SlotsInput) {
+		this._genesisTime = new Date(genesisBlockTimestamp * SEC_IN_MS);
 		this._interval = interval;
 	}
 
-	public getEpochTime(): number {
-		return Math.floor((Date.now() - this._epochTime.getTime()) / SEC_IN_MS);
+	public timeSinceGenesis(): number {
+		return Math.floor((Date.now() - this._genesisTime.getTime()) / SEC_IN_MS);
 	}
 
 	public blockTime(): number {
 		return this._interval;
 	}
 
-	public getRealTime(time: number): number {
-		return (
-			Math.floor(this._epochTime.getTime() / SEC_IN_MS) * SEC_IN_MS +
-			time * SEC_IN_MS
+	public getSlotNumber(timeStamp?: number): number {
+		const elapsedTime = Math.floor(
+			((timeStamp !== undefined ? timeStamp * SEC_IN_MS : Date.now()) -
+				this._genesisTime.getTime()) /
+				SEC_IN_MS,
 		);
-	}
-
-	public getSlotNumber(epochTime?: number): number {
-		const parsedEpochTime =
-			epochTime === undefined ? this.getEpochTime() : epochTime;
 
-		return Math.floor(parsedEpochTime / this._interval);
+		return Math.floor(elapsedTime / this._interval);
 	}
 
 	public getSlotTime(slot: number): number {
-		return slot * this._interval;
+		const slotGensisTimeOffset = slot * this._interval;
```

Comment:
this is returning in `ms`, but `timeSinceGenesis` is returning in `s`, I think it's better to make it consistent

Reply:
Changed.